React/CxxBridge/RCTCxxBridge.mm

```diff
@@ -1060,20 +1060,17 @@ - (void)invalidate
     // wrappers) or we may have invalid pointers still in flight.
     dispatch_group_t moduleInvalidation = dispatch_group_create();
     for (RCTModuleData *moduleData in self->_moduleDataByID) {
+      [moduleData invalidate];
```

Comment:
is the motivation for moving this call before calling the module's `invalidate` method to avoid the race condition?

Reply:
It *could* be moved back to where it was, as long as it's called whether or not `hasInstance` returns YES.

The race condition is fixed by using `m_moduleData.valid` (which is set in the JS thread) instead of `m_bridge.valid` (which is not).